core/domain/translation_domain.py

```diff
@@ -18,9 +18,287 @@
+        """Method to get all the translatable fields in a translatable object.
+
+        Returns:
+            Dict(str, TranslatableContent). Returns the dict containg content_id
+            as key and TranslatableContent as value.
+        """
+        self._translatable_contents = {}
+        self._register_all_translatable_fields()
+        return copy.deepcopy(self._translatable_contents)
+
+    def get_all_contents_which_needs_translations(
+        self,
+        entity_translation: EntityTranslation
+    ) -> List[TranslatableContent]:
+        """Returns a list of TranslatableContent instances which need
+        translation or which need updates in existing translation.
+
+        Args:
+            entity_translation: EntityTranslation. An object storing the
+                existing translations of an entity.
+
+        Returns:
+            list(TranslatableContent). Returns a list of TranslatableContents.
+        """
+        contents_which_need_translation = []
+        content_ids_for_translated_contents = (
+            entity_translation.translations.keys())
+
+        for translatable_item in self.get_translatable_fields().values():
+            # translatable_item is of type TranslatableContent.
+            if translatable_item.content == '':
+                continue
+
+            if (
+                translatable_item.content_id not in
+                content_ids_for_translated_contents
+            ):
+                contents_which_need_translation.append(translatable_item)
+            elif (
+                entity_translation.translations[
+                translatable_item.content_id].needs_update
+            ):
+                contents_which_need_translation.append(translatable_item)
+
+        return contents_which_need_translation
+
+
+class EntityTranslation:
+    """The EntityTranslation represents an EntityTranslationsModel for a given
+    version of an entity in a given language.
+
+    Args:
+        entity_id: str. The id of the corresponding entity.
+        entity_type: EntityTypesSupportingNewTranslationArch. The type
+            of the corresponding entity.
+        entity_version: str. The version of the corresponding entity.
+        language_code: str. The language code for the corresponding entity.
+        translations: dict(str, TranslatedContent). The translated content.
+    """
+
+    def __init__(
+        self,
+        entity_id: str,
+        entity_type: EntityTypesSupportingNewTranslationArch,
+        entity_version: int,
+        language_code: str,
+        translations: Dict[str, TranslatedContent]
+    ):
+        self.entity_id = entity_id
+        self.entity_type = entity_type.value
+        self.entity_version = entity_version
+        self.language_code = language_code
+        self.translations = translations
+
+
+class TranslatableContent:
+    """TranslatablesContent represents a content of a translatable object which
```

Comment:
The first word here doesn't match the class name.

Reply:
Updated.